test/e2e/canary_test.go

```diff
@@ -0,0 +1,176 @@
+// +build e2e
+
+package e2e
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	operatorclient "github.com/openshift/cluster-ingress-operator/pkg/operator/client"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+// TestCanaryRoute tests the ingress canary route
+// and checks that the hello-openshift echo server
+// works as expected.
+func TestCanaryRoute(t *testing.T) {
```

Comment:
Does this provide much coverage over checking for the `CanaryChecksSucceeding` status condition?

Reply:
This test provides much more detailed failure output than the Canary status condition, which would make debugging failed e2e tests easier (wouldn't have to scour through ingress operator / metrics logs when the canary status condition test fails). Also, this test hits the canary endpoint from a pod other than the ingress operator. I guess that isn't really much additional coverage, but I feel it's worth keeping. Thoughts?